src/DataStructures/Python/Matrix.cpp

```diff
@@ -0,0 +1,46 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <boost/python.hpp>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "DataStructures/Matrix.hpp"
+#include "PythonBindings/BoundChecks.hpp"
+#include <boost/python/reference_existing_object.hpp>
+#include <boost/python/return_value_policy.hpp>
+#include <boost/python/tuple.hpp>
+namespace bp = boost::python;
+
+namespace py_bindings {
+void bind_matrix() {
+  // Wrapper for basic Matrix operations
+  bp::class_<Matrix>("Matrix", bp::init<size_t, size_t>())
+    .def("size",
+         +[](const Matrix& M){
```

Comment:
`M` is not snake_case and not descriptive. maybe something like `matrix` would be better, or `self`